### PR TITLE
Corrections release1.0.0 - II

### PR DESCRIPTION
**Pull request by jannikseidelQBiC**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.

# Corrections release 1.0.0 - II
- `fastp` and `kraken2` modules were updated to newest version
- `groupTuple` was extended by `groupKey` to avoid bottlenecks in the pipeline runs, for this an `amount_of_files` entry was added to the meta maps
- `meta.clone()` was replaced with `meta + [changes]`
- `getGenomeAttribute` was moved to `detaxizer.nf` as replacement for logic checking if the genome parameter is in the `igenomes.config` and if the `fasta`attribute is included in this genome entry
- removed doubling of inclusion of `igenomes.config` into `nextflow.config`
- date of release in `CHANGELOG.md` was updated

### CODE REVIEW

File: workflows/detaxizer.nf [hunk 1]
```diff
@@ -38,12 +39,9 @@ include { SUMMARIZER                            } from '../modules/local/summari
 def fasta = false
```
Comment (FriederikeHanssen) [inline workflows/detaxizer.nf:39]: Shouldn't this be an empty channel?

**Review by FriederikeHanssen — COMMENTED** [review]
There are still a few ` groupTuples` left that don't have a size annotated. Are you joining there all channel elements?

**Comment by jannikseidelQBiC** [comment]
> There are still a few ` groupTuples` left that don't have a size annotated. Are you joining there all channel elements?

Yes, as these channels are no bottlenecks and should only be used when all elements are present, I think this is the best way to go.

**Comment by jannikseidelQBiC** [comment]
> > There are still a few ` groupTuples` left that don't have a size annotated. Are you joining there all channel elements?
> 
> Yes, as these channels are no bottlenecks and should only be used when all elements are present, I think this is the best way to go.

Recognized one of the `groupTuple` is not needed and for the others `groupKey` is useful. Will change that and let you know when the full test ran successfully! 👍

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**